src/validation.cpp

```diff
@@ -4626,6 +4626,22 @@ int VersionBitsTipStateSinceHeight(const Consensus::Params& params, Consensus::D
     return VersionBitsStateSinceHeight(::ChainActive().Tip(), params, pos, versionbitscache);
 }
 
+int64_t VersionBitsActivationHeight(const Consensus::Params& params, Consensus::DeploymentPos pos)
+{
+    LOCK(cs_main);
+    ThresholdState state = VersionBitsTipState(params, pos);
+    int64_t since_height = VersionBitsTipStateSinceHeight(params, pos);
+    switch (state) {
+        case ThresholdState::LOCKED_IN:
+            return since_height + params.nMinerConfirmationWindow;
+        case ThresholdState::ACTIVE:
+            return since_height;
+        default:
+            throw std::logic_error("VersionBitsActivationHeight called for non-ACTIVE non-LOCKED-IN deployment");
```

Comment:
done

Reply:
versionbits.cpp::VersionBitsActivationHeight() still has the throw rather than returning -1 as of 41fd9d327f544d0e2f1f419a9ca70282ef5d30d2 though there's a test to see if it returns -1 in rpc/blockchain.cpp

Reply:
fixed